backend/src/routes/hypershift-status.ts

```diff
@@ -0,0 +1,117 @@
+
+import { Http2ServerRequest, Http2ServerResponse } from 'http2'
+import { jsonRequest } from '../lib/json-request'
+import { logger } from '../lib/logger'
+import { respondInternalServerError } from '../lib/respond'
+import { getAuthenticatedToken } from '../lib/token'
+import { getServiceAccountToken } from '../lib/serviceAccountToken'
+
+interface MultiClusterEngine {
+  spec?: {
+    overrides?: {
+      components?: Array<{
+        name: string
+        enabled: boolean
+      }>
+    }
+  }
+}
+
+interface ManagedClusterAddOn {
+  metadata: {
+    name: string
+  }
+  status?: {
+    conditions?: Array<{
+      reason: string
+      status: string
+    }>
+  }
+}
+
+interface ResourceList<T> {
+  items: T[]
+}
+
+function processHypershiftStatus(
+  mceResponse: ResourceList<MultiClusterEngine>,
+  addonResponse: ResourceList<ManagedClusterAddOn>
+): boolean {
+  try {
+    // Find the MultiClusterEngine (should typically be only one)
+    const multiClusterEngine = mceResponse.items?.[0]
+    if (!multiClusterEngine?.spec?.overrides?.components) {
+      return false
+    }
+
+    // Check if hypershift components are enabled
+    const components = multiClusterEngine.spec.overrides.components
+    const hypershift = components.find((component) => component.name === 'hypershift')
+    const hypershiftLocalHosting = components.find((component) => component.name === 'hypershift-local-hosting')
+
+    if (!hypershift?.enabled || !hypershiftLocalHosting?.enabled) {
+      return false
+    }
+
+    // Find the hypershift-addon for the local hub
+    const hypershiftAddon = addonResponse.items.find((addon) => addon.metadata.name === 'hypershift-addon')
+
+    if (!hypershiftAddon?.status?.conditions) {
+      return false
+    }
+
+    // Check if the addon is properly running
+    const isAddonHealthy =
+      hypershiftAddon.status.conditions.find((condition) => condition.reason === 'ManagedClusterAddOnLeaseUpdated')
+        ?.status === 'True'
+
+    return isAddonHealthy
+  } catch (error) {
+    logger.error('Error processing hypershift status:', error)
+    return false
+  }
+}
+
+export async function hypershiftStatus(req: Http2ServerRequest, res: Http2ServerResponse): Promise<void> {
+  const token = await getAuthenticatedToken(req, res)
+  if (!token) {
+    return // getAuthenticatedToken already handles the response
+  }
+
+  const serviceAccountToken = getServiceAccountToken()
+
+  try {
+    // Get the local hub name from query parameter or default to 'local-cluster'
+    const url = new URL(req.url, `http://${req.headers.host}`)
+    const localHubName = url.searchParams.get('hubName') || 'local-cluster'
+
+    // Fetch MultiClusterEngine resources using service account token
+    const mceResponse = await jsonRequest<ResourceList<MultiClusterEngine>>(
+      process.env.CLUSTER_API_URL + '/apis/multicluster.openshift.io/v1/multiclusterengines',
+      serviceAccountToken
+    )
```

Comment:
Please see ./backend/src/lib/multi-cluster-engine.ts and ./backend/src/lib/multi-cluster-hub.ts

We already have a function for fetching the MCE (and a type definition for it), and I think we could add a function there for getting its components similar to what Kike did for MCH.

Similarly, the code for fetching ManagedClusterAddOns could be in its own file.